validate/validate.go

```diff
@@ -68,6 +68,7 @@ type Validator struct {
 // NewValidator creates a Validator
 func NewValidator(spec *rspec.Spec, bundlePath string, hostSpecific bool, platform string) Validator {
 	if hostSpecific && platform != runtime.GOOS {
+		logrus.Warn("platform is overwriteen by the host platform")
```

Comment:
typo: “overwriteen” → “overwritten”, and again below.

In #414 I was calling for this situation to be a *fatal* “we can't do that on this host” error, not the “we can't do what you asked for so we're going to do something else instead” warning you have here ;).

Reply:
The role of `host-specific` is to execute host-specific validations, so I think I should change the value of the `platform`  and output the warning,  then continue to execute, instead of reporting an error and terminating the execution.

Reply:
> ...so I think I should change the value of the platform and output the warning, then continue to execute...

But once we fix `--platform`, any mismatch will be because the caller specifically requested host-specific validation and specifically requested the wrong platform.  I think that's a fatally-wrong request.

Reply:
>any mismatch will be because the caller specifically requested host-specific validation and specifically requested the wrong platform.

In my opinion, warning is enough. @liangchenye @Mashimiao What do you think?

Reply:
I'm OK to change default value of --platform.
About value of platform doesn't equal to runtime.GOOS, I think the simplest way is to return an error and give user the reason.

Reply:
Fine, updated.